components/brave_component_updater/browser/component_contents_verifier.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_component_updater/browser/component_contents_verifier.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/check_is_test.h"
+#include "base/files/file_util.h"
+#include "base/no_destructor.h"
+#include "base/task/thread_pool.h"
+
+namespace brave_component_updater {
+
+ComponentContentsAccessor::ComponentContentsAccessor(
+    const base::FilePath component_root)
+    : component_root_(component_root) {}
+
+bool ComponentContentsAccessor::IsComponentSignatureValid() const {
+  return true;
+}
+
+bool ComponentContentsAccessor::VerifyContents(
+    const base::FilePath& relative_path,
+    base::span<const uint8_t> contents) {
+  return true;
+}
+
+const base::FilePath& ComponentContentsAccessor::GetComponentRoot() const {
+  return component_root_;
+}
+
+std::optional<std::string> ComponentContentsAccessor::GetFileAsString(
+    const base::FilePath& relative_path) {
+  std::string contents;
+  if (!base::ReadFileToString(GetComponentRoot().Append(relative_path),
+                              &contents)) {
+    return std::nullopt;
+  }
+  if (!VerifyContents(relative_path, base::as_byte_span(contents))) {
+    return std::nullopt;
+  }
+
+  return contents;
+}
+
+std::optional<std::vector<uint8_t>> ComponentContentsAccessor::GetFileAsBytes(
+    const base::FilePath& relative_path) {
+  auto contents =
+      base::ReadFileToBytes(GetComponentRoot().Append(relative_path));
+  if (!contents) {
+    return std::nullopt;
+  }
+  if (!VerifyContents(relative_path, base::as_byte_span(*contents))) {
+    return std::nullopt;
+  }
+  return std::move(*contents);
+}
+
+ComponentContentsVerifier::ComponentContentsVerifier() = default;
+ComponentContentsVerifier::~ComponentContentsVerifier() = default;
+
+// static
+void ComponentContentsVerifier::Setup(
+    ComponentContentsVerifier::ComponentContentsAccessorFactory cca_factory) {
+  if (GetInstance()->cca_factory_) {
+    CHECK_IS_TEST();
+  }
+  GetInstance()->cca_factory_ = std::move(cca_factory);
+}
+
+// static
+ComponentContentsVerifier* ComponentContentsVerifier::GetInstance() {
+  static base::NoDestructor<ComponentContentsVerifier> instance;
+  return instance.get();
+}
+
+void ComponentContentsVerifier::CreateContentsAccessor(
+    const base::FilePath& component_root,
+    base::OnceCallback<void(scoped_refptr<ComponentContentsAccessor>)>
+        on_created) {
+  CHECK(cca_factory_);
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE, {base::MayBlock()},
```

Comment:
Does the default priority matches the priority used by adblock components?
I thought it should be at least `USER_VISIBLE`..

Reply:
Not it is created on VerifyInstallation with UV priority.